dlt/sources/rest_api/typing.py

```diff
@@ -200,14 +200,18 @@ class OAuth2ClientCredentialsConfig(AuthTypeConfig, total=False):
 ]
 
 
-class ClientConfig(TypedDict, total=False):
-    base_url: str
+class BaseClientConfig(TypedDict, total=False):
+    base_url: Optional[str]
     headers: Optional[Dict[str, str]]
     auth: Optional[AuthConfig]
     paginator: Optional[PaginatorConfig]
     session: Optional[Session]
 
 
+class ClientConfig(BaseClientConfig, total=False):
+    base_url: str  # type: ignore[misc]
```

Comment:
Typing overrides are currently not yet supported (see https://github.com/python/mypy/issues/7435) thus the ignore here.
For the optional dict in each endpoint we need to ensure that all keys are optional.